File: purchases/src/main/java/com/revenuecat/purchases/interfaces/MakePurchaseListener.java
//  Purchases
//
//  Copyright © 2019 RevenueCat, Inc. All rights reserved.
//

package com.revenuecat.purchases.interfaces;

import androidx.annotation.NonNull;

import com.android.billingclient.api.Purchase;
import com.revenuecat.purchases.CustomerInfo;

/**
 * Interface to be implemented when making purchases.
 * @deprecated
 * <p> Use {@link PurchaseCallback}, which returns a
 * {@link com.revenuecat.purchases.models.StoreTransaction} instead of a {@link Purchase}.</p>
 * <p>You can use `MakePurchaseListener.toPurchaseCallback()` in Kotlin for an easy migration.</p>
 */
@Deprecated
public interface MakePurchaseListener extends PurchaseErrorListener {
    /**
     * Will be called after the purchase has completed
     * @param purchase Purchase object for the purchased product.
     * @param customerInfo Updated [CustomerInfo].
     */
    void onCompleted(@NonNull Purchase purchase, @NonNull CustomerInfo customerInfo);

}

File: purchases/src/main/java/com/revenuecat/purchases/interfaces/MakePurchaseListener.kt
//  Purchases
//
//  Copyright © 2019 RevenueCat, Inc. All rights reserved.
//
package com.revenuecat.purchases.interfaces

import com.android.billingclient.api.Purchase
import com.revenuecat.purchases.PurchaserInfo

/**
 * Interface to be implemented when making purchases.
 */
@Deprecated(
    """
       Replace with PurchaseCallback, which returns a StoreTransaction instead of a Purchase, and a CustomerInfo 
       instead of a PurchaserInfo. You can use `MakePurchaseListener.toPurchaseCallback()` in Kotlin for an 
       easy migration 
    """,
    replaceWith = ReplaceWith("PurchaseCallback")
)
interface MakePurchaseListener : PurchaseErrorListener {
    /**
     * Will be called after the purchase has completed
     * @param purchase Purchase object for the purchased product.
     * @param purchaserInfo Updated [PurchaserInfo].
     */
    fun onCompleted(purchase: Purchase, purchaserInfo: PurchaserInfo)
}
